### DISCUSSION TITLE
Issue Can Speed & Frame Sender

### DISCUSSION THREAD

**Discussion by LukasUp** [collin80/SavvyCAN#329] [Q&A]
Hello thanks for this nice application.
I have some troubles with SavvyCan and a kvaser interface und ubuntu. Sometimes i'm not able to change the speed to 500kbps. It always stays at 200. And 2nd after i have changed a byte in the frame sender, the SavvyCan says it is not able to send a CAN FD message but its not a can FD message? After removing all grids it works perfect again.

How can i solve this issue

Issue 1 set speed
![Bildschirmfoto zu 2021-04-11 16-26-12](https://user-images.githubusercontent.com/61162331/114308873-96048900-9ae5-11eb-8014-902ac0d9eea0.png)

Issue 2 send Can Frame
![Bildschirmfoto zu 2021-04-11 16-37-34](https://user-images.githubusercontent.com/61162331/114308863-90a73e80-9ae5-11eb-92ed-a11beb39e68e.png)

**Comment by collin80** [comment]
Socketcan doesn't tend to like applications trying to set the speed. At least it used to be totally impossible to do that - you'd have to set the speed from the command line when setting up the socketcan interface. I think newer versions of QT might have added some functionality to attempt it but this is probably still hit or miss. The best approach is to set the speed from the command line and then just tell SavvyCAN which interface to use and don't mess with the settings.

As for the CAN FD issue - I will have to look into that more deeply. I did add some preliminary support for CAN FD to SavvyCAN but it shouldn't have been trying to send FD frames. Apparently I must have accidentally done so.

I'll test both issues. I have a kvaser leaf light and use Ubuntu so I can replicate these things.

**Reply by LukasUp** [reply]
Thx for your answer. 
Up to now it works really fine . It's a great software. 
For sending requests. Is it possible so set an additional filter to the ID? For ex. 7ED xx 62 07 xx xx xx xx xx. It should trigger the ID and the 1st and 2nd byte of a can frame.
